File: VectoCommon/AdvancedAuxiliaryInterfaces/IFuelConsumptionMap.vb
Imports TUGraz.VectoCommon.Utils

Public Interface IFuelConsumptionMap
	''' <summary>
	''' 
	''' </summary>
	''' <param name="torque"></param>
	''' <param name="angularVelocity"></param>
	''' <returns></returns>
	''' <remarks></remarks>
	Function GetFuelConsumption(torque As NewtonMeter, angularVelocity As PerSecond) As KilogramPerSecond
End Interface

File: VectoCommon/VectoCommon/Models/IFuelConsumptionMap.cs
using System.Diagnostics;
using TUGraz.VectoCommon.Utils;

namespace TUGraz.VectoCommon.Models
{
    public interface IFuelConsumptionMap
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="torque"></param>
        /// <param name="angularVelocity"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        KilogramPerSecond GetFuelConsumptionValue(NewtonMeter torque, PerSecond angularVelocity);

        FuelConsumptionResult GetFuelConsumption(NewtonMeter currentStateEngineTorque, PerSecond avgEngineSpeed, bool allowExtrapolation = false);
    }

    [DebuggerDisplay("{Value} (extrapolated: {Extrapolated})")]
    public class FuelConsumptionResult
    {
        public KilogramPerSecond Value;
        public bool Extrapolated;
    }
}
